spec.html

```diff
@@ -37627,29 +37597,55 @@ <h1>Array.prototype.findLast ( _predicate_ [ , _thisArg_ ] )</h1>
+              _direction_: ~ascending~ or ~descending~,
+              _predicate_: an ECMAScript language value,
+              _thisArg_: an ECMAScript language value,
+            ): either a normal completion containing a Record with fields [[Index]] (an integral Number) and [[Value]] (an ECMAScript language value) or a throw completion
+          </h1>
+          <dl class="header">
+            <dt>description</dt>
+            <dd>
+              <p>_O_ should be an array-like object or a TypedArray. This operation calls _predicate_ once for each element of _O_, in either ascending index order or descending index order (as indicated by _direction_), until it finds one where _predicate_ returns a value that coerces to *true*. At that point, this operation returns a Record that gives the index and value of the element found. If no such element is found, this operation returns a Record that specifies *-1*<sub>𝔽</sub> for the index and *undefined* for the value.</p>
+              <p>_predicate_ should be a function. When called for an element of the array, it is passed three arguments: the value of the element, the index of the element, and the object being traversed. Its return value will be coerced to a Boolean value.</p>
+              <p>_thisArg_ will be used as the *this* value for each invocation of _predicate_.</p>
+              <p>This operation does not directly mutate the object on which it is called, but the object may be mutated by the calls to _predicate_.</p>
+              <p>The range of elements processed is set before the first call to _predicate_, just before the traversal begins. Elements that are appended to the array after this will not be visited by _predicate_. If existing elements of the array are changed, their value as passed to _predicate_ will be the value at the time that this operation visits them. Elements that are deleted after traversal begins and before being visited are still visited and are either looked up from the prototype or are *undefined*.</p>
+            </dd>
+          </dl>
+          <emu-alg>
+            1. If IsCallable(_predicate_) is *false*, throw a *TypeError* exception.
+            1. If _direction_ is ~ascending~, then
+              1. Let _indices_ be a List of the integers in the interval from 0 (inclusive) to _len_ (exclusive), in ascending order.
+            1. Else,
+              1. Let _indices_ be a List of the integers in the interval from 0 (inclusive) to _len_ (exclusive), in descending order.
+            1. For each integer _k_ of _indices_, do
+              1. Let _Pk_ be ! ToString(𝔽(_k_)).
+              1. NOTE: If _O_ is a TypedArray, the following invocation of Get will return a normal completion.
+              1. Let _kValue_ be ? Get(_O_, _Pk_).
+              1. Let _testResult_ be ToBoolean(? Call(_predicate_, _thisArg_, « _kValue_, 𝔽(_k_), _O_ »)).
+              1. If _testResult_ is *true*, return the Record { [[Index]]: 𝔽(_k_), [[Value]]: _kValue_ }.
```

Comment:
```suggestion
              1. Let _testResult_ be ? Call(_predicate_, _thisArg_, « _kValue_, 𝔽(_k_), _O_ »).
              1. If ToBoolean(_testResult_) is *true*, return the Record { [[Index]]: 𝔽(_k_), [[Value]]: _kValue_ }.
```

Reply:
I think this suggestion is slightly more readable, but I'm not sure.

One argument against would be to keep consistency with `{Array,%TypedArray%}.p.{every,some}`, but I'd say that's pretty weak.

Reply:
I agree that the suggested change improves readability.

Reply:
I agree it is very slightly more readable, so I am (very slightly) in favor of this change despite the inconsistency.

Reply:
Added a commit with this change.